package/upgrade/upgrade_manifests.sh

```diff
@@ -316,21 +303,69 @@ EOF
   # Sleep 1 sec to ensure force_update_time is always in the past
   sleep 1
 
-  cat > catalog_cluster_repo.yaml << EOF
+  cat >catalog_cluster_repo.yaml <<EOF
 spec:
   forceUpdate: "$force_update_time"
 EOF
   kubectl patch clusterrepos.catalog.cattle.io harvester-charts --patch-file ./catalog_cluster_repo.yaml --type merge
 
-  until [ $(get_cluster_repo_index_download_time epoch) -ge $(date -d"${force_update_time}" +%s) ]
-  do
+  until [ $(get_cluster_repo_index_download_time epoch) -ge $(date -d"${force_update_time}" +%s) ]; do
     echo "Waiting for cluster repo catalog index update..."
     sleep 5
   done
 }
 
-upgrade_harvester()
-{
+upgrade_network(){
+  [[ $UPGRADE_PREVIOUS_VERSION != "v1.0.3" ]] && return
+
+  shutdown_all_vms
+  wait_all_vms_shutdown
+  modify_nad_bridge
+  delete_canal_flannel_iface
+}
+
+wait_all_vms_shutdown() {
+    local vm_count="$(get_all_running_vm_count)"
+
+    until [ "$vm_count" = "0" ]
+    do
+      echo "Waiting for VM shutdown...($vm_count left)"
+      sleep 5
+      vm_count="$(get_running_vm_count)"
```

Comment:
`get_running_vm_count` is defined in `upgrade_node.sh`. Maybe we should move it into `lib.sh`?
<img width="878" alt="Screen Shot 2022-10-11 at 12 29 33" src="https://user-images.githubusercontent.com/1827717/194997398-6cb72701-8903-417c-bc31-ebd11796bacf.png">

Reply:
I think this should be "get_all_running_vm_count" rather than "get_running_vm_count"